.github/workflows/release.yml

```diff
@@ -0,0 +1,96 @@
+name: release
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - main
+      - dev-*
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-binaries:
+    runs-on: ubuntu-latest
+    if: github.repository == 'numaproj-labs/numaplane'
+    name: Build binaries
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+
+  build-push-linux-multi:
+    name: Build & push linux/amd64 and linux/arm64
+    needs: [ build-binaries ]
+    runs-on: ubuntu-latest
+    if: github.repository == 'numaproj-labs/numaplane'
+    strategy:
+      matrix:
+        target: [ numaplane ]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20"
+      - name: Setup Node-Cache
+        uses: actions/cache@v4
+        with:
+          path: ui/node_modules
+          key: ${{ runner.os }}-node-dep-v1-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Registry Login
+        uses: docker/login-action@v2
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAYIO_USERNAME }}
+          password: ${{ secrets.QUAYIO_PASSWORD }}
+
+      - name: set Version
+        id: version
+        run: |
+          tag=$(basename $GITHUB_REF)
+          if [ $tag = "main" ]; then
+            tag="latest"
+          fi
+          echo "::set-output name=VERSION::$tag"
+      - name: Container build and push with arm64/amd64
+        run: make image docker-push IMAGE_FULL_PATH=${{ secrets.QUAYIO_ORG }}/numaplane:${{ steps.version.outputs.VERSION }} CONTAINER_TOOL=docker
```

Comment:
I'm not familiar with "::set-output" - is doing that above setting `steps.version.outputs.VERSION`? Was there a reason to do that instead of using the `$GITHUB_OUTPUT` environment variable that numaflow is using, which seems to be mentioned in the [tutorial](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs) on Github Actions Outputs? (I'm seeing something about set-output being [deprecated](https://github.com/orgs/community/discussions/35994).)